code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -1127,6 +1139,11 @@ All effects don't start immediately, but rather get worse over time; the rate is
 		. = 1
 	return ..() || .
 
+/datum/reagent/consumable/ethanol/hearty_punch/on_mob_delete(mob/living/M)
+	if(active)
+		active = FALSE
```

Comment:
for what purpose

Reply:
In case the reagents run out before the active state is removed - wasn't sure if it was necessary, but I figured I'd have it just in case some stupid stuff happens.

Reply:
reagents when inside reagent holders are instances that are deleted when they deplete so this shouldn't be necessary, a reagent shouldn't tick after it's been removed unless there's a bigger problem somewhere else